src/monitor/node_active_protocol.c

```diff
@@ -554,21 +554,6 @@ JoinAutoFailoverFormation(AutoFailoverFormation *formation,
 
 			initialState = REPLICATION_STATE_WAIT_STANDBY;
 
-			/*
```

Comment:
The comment made me think on the PR again. 
We have [this](https://github.com/citusdata/pg_auto_failover/blob/master/src/monitor/group_state_machine.c#L314-L338) code block in group_state_machine, which is the place that we mark nodes primary/secondary once A secondary  catches up.  
- If we have the other secondary catching up concurrently, the same block will transition to `primary->primary`, `catching up -> secondary`, which is good
-  Even if we add a new standby, the primary switch back to  `wait_primary` and things will  work as expected

Well, this scenario is hard to test locally because even if I have 100GB data, it catches up so fast that I cannot add new standby while others are in progress. 

I wanted to write my understanding here in case this pops-up any edge cases on your hear.

Reply:
> If we have the other secondary catching up concurrently, the same block will transition to primary->primary, catching up -> secondary, which is good

This  is slightly confusing from the user perspective though. Because now we could have `primary` and `catching up` at the same  time

Reply:
Have a look at https://github.com/citusdata/pg_auto_failover/pull/398 where we're creating a tmux script that automates creating a formation. If you remove the `sleep` calls in there, you should be able to generate this race condition more easily.

Otherwise, we have added PRIMARY in the list of states you're showing in that code block in the multiple standby PR, because now we can have standby showing up when the primary is already back to being a primary (failover, report_lsn, back to normal, and a standby was not connecting for reasons and is back just now, would be a scenario, IIRC). So I think we're good here, in particular thanks to the block of code you have highlighted, yes.